src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs

```diff
@@ -85,21 +97,154 @@ public static bool IsValidFormatArg(string formatString, CultureInfo formatCultu
                 return false;
             }
 
-            textFormatArgs.HasDateTimeFmt = textFormatArgs.FormatArg.IndexOfAny(new char[] { 'm', 'M', 'd', 'D', 'y', 'Y', 'h', 'H', 's', 'S', 'a', 'A', 'p', 'P' }) >= 0;
-            textFormatArgs.HasNumericFmt = textFormatArgs.FormatArg.IndexOfAny(new char[] { '0', '#' }) >= 0;
-            if (textFormatArgs.HasDateTimeFmt && textFormatArgs.HasNumericFmt)
+            bool hasNumericCharacters = false;
+            int decimalPointIndex = -1;
+            int sectionCount = 0;
+            var formatStr = textFormatArgs.FormatArg;
+            bool hasColonWithNum = false;
+
+            //Block "gen"/"general"
```

Comment:
>"gen"/"general" [](http://example.com/codeflow?start=20&length=15)

They keyword in Excel is "general" - not "gen". We can search for that explicitly

Reply:
updated here.